### PR TITLE
Fix panther-log-analysis-iam.yml

### PR DESCRIPTION
**Pull request by giorgosp**
## Background

Fix panther-log-analysis-iam.yml

## Changes

- Properly check for `IsGenerated`
- Refactor for better readability

## Testing

- Deployed stack twice: Once with populated `Mappings` and once by setting parameters on the AWS console.
- Used the role with `awscli` to fetch events from a bucket (with and without prefix in the key)

### CODE REVIEW

**Comment by kostaspap** [comment]
> * Deployed stack twice: Once with populated `Mappings` and once by setting parameters on the AWS console.

I guess you mean that in the first case, you downloaded the CFN template from the UI and deployed it manually and in the second, you deployed the template using the redirect link from the UI?

File: deployments/auxiliary/cloudformation/panther-log-analysis-iam.yml [hunk 1]
```diff
@@ -97,19 +97,26 @@ Resources:
...
-                    - Bucket: !FindInMap [PantherParameters, S3Bucket, Value]
-                - Effect: Allow
-                  Action: s3:GetBucketLocation
-                  Resource: !Sub 'arn:aws:s3:::${S3Bucket}'
+              - Effect: Allow
+                Action: s3:GetBucketLocation
+                Resource: !Sub
+                  - 'arn:aws:s3:::${Bucket}'
+                  - Bucket:
+                      !If [
```
Comment (giorgosp) [inline deployments/auxiliary/cloudformation/panther-log-analysis-iam.yml:105]: I had this in one line and was quite nice and clean but `mage fmt` formatted it into multiline :(

**Review by kostaspap — APPROVED** [review]
Looks good! Once this is merged, we need to make sure to publish it to our buckets

**Comment by giorgosp** [comment]
> > * Deployed stack twice: Once with populated `Mappings` and once by setting parameters on the AWS console.
> 
> I guess you mean that in the first case, you downloaded the CFN template from the UI and deployed it manually and in the second, you deployed the template using the redirect link from the UI?

The template is not deployed on the s3 bucket so it was easier for me to do manually what Panther does.
In the "generated" case Panther just replaces the fields inside the `Mappings` section. I replaced them manually and
uploaded the template manually.

In the second case I uploaded a template without populating anything and set the template parameters on AWS UI. This emulated the "redirect link" flow of Panther

**Comment by giorgosp** [comment]
@s0l0ist @austinbyers Could someone re-deploy this template for me please?

**Comment by austinbyers** [comment]
> Could someone re-deploy this template for me please? Templates for 1.15 / 1.15.1 / 1.15.2 should be replaced

Interesting, then we need to sync this change to the release branch

**Pull request merged**